### ISSUE TITLE
Correctness of flip formula

### ISSUE THREAD

**Issue by ratthachat** [ultralytics/yolov5#248]
First of all thanks so much for making this Yolo V5! It's now the best model in Kaggle Wheat detection : https://www.kaggle.com/c/global-wheat-detection

Trying to modify TTA, I found you use the following lr-flip formula for bbox : 

In `models/yolo.py`
`y[1][..., 0] = img_size[1] - y[1][..., 0]  # flip lr ---- Equation (1)` 

I have checked on Yolo V3 and it's the same, so I am sure the formula is correct.

However, in my understanding we should change both `xmin` and `xmax `for lr-flip but the formula above only change `xmin`.
For examples, the famous albumentations formula is :

https://github.com/albumentations-team/albumentations/blob/master/albumentations/augmentations/functional.py
In `def bbox_hflip`
`new_xmin, new_ymin, new_xmax, new_ymax = 1 - x_max, y_min, 1 - x_min, y_max`

Or explained intuitively here :  https://blog.paperspace.com/data-augmentation-for-bounding-boxes/

So my question is how can the above equation (1) is correct ??
It should be

`y[1][..., 2] = img_size[1] - y[1][..., 0]  # flip lr new xmax` 
`y[1][..., 0] = img_size[1] - y[1][..., 2]  # flip lr new xmin` 

**EDIT** : I got it. Yolo use `xcenter, ycenter, w, h`, so it explains Equation (1) . Close the issue.

**Comment by glenn-jocher** [comment]
Yes, xywh format.

**Comment by glenn-jocher** [comment]
@ratthachat see PR #3882 for a proposed automatic Albumentations integration.

**Comment by glenn-jocher** [comment]
@ratthachat good news 😃! Your original issue may now be fixed ✅ in PR #3882. This PR implements a [YOLOv5](https://github.com/ultralytics/yolov5) 🚀 + [Albumentations](https://github.com/albumentations-team/albumentations) integration. The integration will automatically apply Albumentations transforms during YOLOv5 training if `albumentations>=1.0.0` is installed in your environment.

## Get Started

To use albumentations simply `pip install -U albumentations` and then update the augmentation pipeline as you see fit in the `Albumentations` class in `yolov5/utils/augmentations.py`. Note these Albumentations operations run **in addition to** the YOLOv5 **hyperparameter** augmentations, i.e. defined in [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyps/hyp.scratch.yaml).

```python
class Albumentations:
    # YOLOv5 Albumentations class (optional, used if package is installed)
    def __init__(self):
        self.transform = None
        try:
            import albumentations as A
            check_version(A.__version__, '1.0.0')  # version requirement

            self.transform = A.Compose([
                A.Blur(p=0.1),
                A.MedianBlur(p=0.1),
                A.ToGray(p=0.01)],
                bbox_params=A.BboxParams(format='yolo', label_fields=['class_labels']))

            logging.info(colorstr('albumentations: ') + ', '.join(f'{x}' for x in self.transform.transforms))
        except ImportError:  # package not installed, skip
            pass
        except Exception as e:
            logging.info(colorstr('albumentations: ') + f'{e}')

    def __call__(self, im, labels, p=1.0):
        if self.transform and random.random() < p:
            new = self.transform(image=im, bboxes=labels[:, 1:], class_labels=labels[:, 0])  # transformed
            im, labels = new['image'], np.array([[c, *b] for c, b in zip(new['class_labels'], new['bboxes'])])
        return im, labels
```

## Example Result

Example `train_batch0.jpg` on COCO128 dataset with Blur, MedianBlur and ToGray. See the YOLOv5 [Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) to reproduce: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

![train_batch0](https://user-images.githubusercontent.com/26833433/124400879-ff331b80-dd25-11eb-9b67-fe85ac4ca104.jpg)


# Update

To receive this YOLOv5 update:

- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
